Magento2/Sniffs/Legacy/ObsoleteAclSniff.php

```diff
@@ -0,0 +1,79 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento2\Sniffs\Legacy;
+
+use DOMDocument;
+use PHP_CodeSniffer\Files\File;
+use PHP_CodeSniffer\Sniffs\Sniff;
+use SimpleXMLElement;
+
+/**
+ * Test to find obsolete acl declaration
+ */
+class ObsoleteAclSniff implements Sniff
+{
+    private const WARNING_OBSOLETE_ACL_STRUCTURE = 'ObsoleteAclStructure';
+
+    /**
+     * @inheritdoc
+     */
+    public function register(): array
+    {
+        return [
+            T_INLINE_HTML
+        ];
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function process(File $phpcsFile, $stackPtr)
+    {
+        $xml = simplexml_load_string($this->getFormattedXML($phpcsFile));
+        $foundElements = $xml->xpath('/config/acl/*[boolean(./children) or boolean(./title)]');
+        foreach ($foundElements as $element) {
+            if (!$this->elementIsCurrentlySniffedLine($element, $stackPtr)) {
```

Comment:
I think a better approach now is to return at the beginning of `process` id `$stackPtr` is not `0`, this way, you'll only enter in the XML file once and won't need to use `elementIsCurrentlySniffedLine`. Take a look at `WidgetXMLSniff.php`